src/Adapter/Image/ImageRetriever.php

```diff
@@ -48,9 +49,15 @@ class ImageRetriever
      */
     private $link;
 
-    public function __construct(Link $link)
+    /**
+     * @var AvifExtensionChecker
+     */
+    private $avifExtensionChecker;
+
+    public function __construct(Link $link, AvifExtensionChecker $avifExtensionChecker = null)
```

Comment:
As `Add an optional constructor parameter at the end of the args list` is allowed in minor versions, you could go further and inject the Configuration service there.

Also, I'm not entirely sure if this is the right way if this is a legacy class, and we could do this much more simpler for the time being.